src/Runner.Sdk/Util/UrlUtil.cs

```diff
@@ -8,7 +8,8 @@ public static bool IsHostedServer(UriBuilder gitHubUrl)
         {
             return string.Equals(gitHubUrl.Host, "github.com", StringComparison.OrdinalIgnoreCase) ||
                 string.Equals(gitHubUrl.Host, "www.github.com", StringComparison.OrdinalIgnoreCase) ||
-                string.Equals(gitHubUrl.Host, "github.localhost", StringComparison.OrdinalIgnoreCase);
+                string.Equals(gitHubUrl.Host, "github.localhost", StringComparison.OrdinalIgnoreCase) ||
+                gitHubUrl.Host.EndsWith(".ghe.com", StringComparison.OrdinalIgnoreCase);
```

Comment:
something like
```C#
var takeOverGhe = StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable("GITHUB_ACTIONS_RUNNER_TAKEOVER_GHE_DOTCOM"));
return string.Equals(gitHubUrl.Host, "github.com", StringComparison.OrdinalIgnoreCase) ||
                string.Equals(gitHubUrl.Host, "www.github.com", StringComparison.OrdinalIgnoreCase) ||
                string.Equals(gitHubUrl.Host, "github.localhost", StringComparison.OrdinalIgnoreCase) ||
                (!takeOverGhe && gitHubUrl.Host.EndsWith(".ghe.com", StringComparison.OrdinalIgnoreCase));
```

Reply:
in case GHES customers for whatever reason used `.ghe.com`, they can `export GITHUB_ACTIONS_RUNNER_TAKEOVER_GHE_DOTCOM=1` and make their things keep working.